myriad-scheduler/src/main/java/org/apache/myriad/scheduler/fgs/YarnNodeCapacityManager.java

```diff
@@ -148,9 +149,9 @@ private void removeYarnTask(RMContainer rmContainer) {
     }
   }
 
-  private Node retrieveNode(RMContainer container) {
+  private Optional<Node> retrieveNode(RMContainer container) {
     String hostname = container.getContainer().getNodeId().getHost();
-    return nodeStore.getNode(hostname);
+    return Optional.fromNullable(nodeStore.getNode(hostname));
   }
```

Comment:
This makes more sense as it's forcing the consumer to think about the empty case.